src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -196,6 +166,15 @@ private TagSpanIntervalTree<TTag> GetTagTree(ITextSnapshot snapshot, ImmutableDi
                     : new TagSpanIntervalTree<TTag>(snapshot.TextBuffer, _dataSource.SpanTrackingMode);
             }
 
+            private async Task ProcessEventsAsync(ImmutableArray<bool> events, CancellationToken cancellationToken)
+            {
+                // Can only have at most a single `true` and `false` value in this as we are deduping these notification values.
+                Contract.ThrowIfTrue(events.Length > 2);
```

Comment:
I can see that there will only be two events because of the equality comparer passed into the work queue.  But is there a real need for this contract?  Even if there were more events it doesn't seem like it would be a problem since we only care if one of them is true

Reply:
> But is there a real need for this contract?

Well, the contract just proves i'm sane.  If it were to fire, i'd want to know, as it would indicate something broken in the AsyncBatchignWorkQueue.  It's an open philosophical questiuon if any contracts are truly required.  However, i still like them :)

> Even if there were more events it doesn't seem like it would be a problem since we only care if one of them is true

True.  However, it would mean something went terribly wrong with AsyncBatchignWorkQueue.  And that is something we'd need to know about to fix since so much depends on it.